scripts/convert_dat.py

```
@@ -74,35 +96,46 @@ def read_fixed_width_file(
             ]
         )
         .drop(["full_str"])
-        .cast({pl.selectors.by_name(NON_ZEROES + ZEROES): pl.Float32})
+        # .cast({pl.selectors.by_name(NON_ZEROES + ZEROES): pl.Float32})
     )
 
     return df
 
 
-def convert(infile: str, width: int = 26) -> pl.DataFrame:
+def convert(infile: str, width: int = 26, archive = False) -> pl.DataFrame:
     """Reads in a file in fwf and returns a polars dataframe.
 
     Args:
         infile: filename to read
         width: column width in fwf
     """
-    with open(infile) as f:
-        header = f.readline().split()
+    if archive:
+        return read_h5(infile)
+    else:
+        with open(infile) as f:
+            header = f.readline().split()
 
-    return read_fixed_width_file(infile, header, skip_rows=1, width=width)
+        return read_fixed_width_file(infile, header, skip_rows=1, width=width)
 
 
-def aggregate_df(data):
+def aggregate_df(data, archive=False):
     """Returns a dataframe aggregated from every second to every hour.
 
     Args:
         data: the dataframe to aggregate
     """
+    logging.info("Aggregating df for firestore")
     # add hour and filter to only good data (no alarm status, not warming up)
+    if not archive:
+        data = data.with_columns(nans=pl.all_horizontal(data!="")).filter(pl.col("nans"))
+        
+    try:
+        data = data.cast({pl.selectors.by_name(NON_ZEROES + ZEROES): pl.Float32})
+    except Exception as e:
+        logging.error(f"Could not cast data to float! {e}")
```

You still need to raise the error afterwards right?